src/user/user.service.ts

```diff
@@ -0,0 +1,58 @@
+import { Injectable, NotFoundException } from '@nestjs/common';
+import { CreateUserDto } from './dto/create-user.dto';
+import { UpdateUserDto } from './dto/update-user.dto';
+import { v4 as uuidv4 } from 'uuid';
+import { UsersDataBase } from './users-storage';
+import { User } from './entities/user.entity';
+import { SortArray } from 'src/helpers/sortArray';
+
+@Injectable()
+export class UserService {
+  constructor(private readonly usersDataBase: UsersDataBase) {}
+
+  create(createUserDto: CreateUserDto): User {
+    const newUser: User = {
+      id: uuidv4(),
+      ...createUserDto,
+      age: +createUserDto.age,
+      isDeleted: false,
+    };
+    return this.usersDataBase.create(newUser);
+  }
+
+  findAll(loginSubstring: string, limit: number) {
```

Comment:
May be we should have ability to make GET requests without using Query params? may be we can implement using default values of  loginSubstring & limit here

Reply:
This is a really important note, fixed this bug.